### PR TITLE
 feat(trailhead): add new designs to the sign_up_password page 

### PR DESCRIPTION
**Pull request by shane-tomlinson**
To handle whether the marketing email checkbox is displayed
on the sign_up* pages and should propagate the checkbox value,
the email-opt-in-mixin was modified to provide two methods where
one was assumed previously:

* isEmailOptInEnabled returns `true` if email opt-in is enabled for the locale
* isEmailOptInVisible returns `true` if email opt-in is visible on the page

Whether the checkbox's value is propagated is gated on isEmailOptInVisible.
If not visible, do not propagate the value.

This allows the mixin to behave correctly for both the legacy flow
where the checkbox is on the sign_up* pages and the trailhead
flow where it is displayed on CWTS.

extraction from #1098
Builds on #1158

Spec'd at https://mozilla.invisionapp.com/share/3GRZJ9J65A9#/screens/362776357

### Desktop
Screenshot 2019-05-20 at 19 59 31

### Mobile
Screenshot 2019-05-20 at 19 59 38

@mozilla/fxa-devs - r?

### CODE REVIEW

File: packages/fxa-content-server/app/styles/_base.scss [hunk 1]
```diff
@@ -232,3 +254,42 @@ a[data-visible-url^='http']::after {
...
 }
+
+.step {
+  background-position: center center;
+  background-repeat: no-repeat;
+  background-size: contain;
+  display: block;
+  margin: 0 auto 20px;
+
+
```
Comment (vbudhram) [inline packages/fxa-content-server/app/styles/_base.scss]: nit: extra space

File: packages/fxa-content-server/app/scripts/views/sign_up.js [hunk 2]
```diff
@@ -271,12 +271,16 @@ var View = FormView.extend({
...
-      email: this.getElementValue('.email'),
-      needsOptedInToMarketingEmail: this.hasOptedInToMarketingEmail()
+      email: this.getElementValue('.email')
     });
 
+    if (this.isEmailOptInVisible()) {
+      account.set('needsOptedInToMarketingEmail', this.hasOptedInToMarketingEmail());
+    }
+
+
```
Comment (vbudhram) [inline packages/fxa-content-server/app/scripts/views/sign_up.js]: nit: extra space

**Comment by vbudhram** [comment]
Strange, mocha test pass locally but circle is complaining

```
× firefox on linux 4.15.0-1035-aws - mocha tests - run the mocha tests (300.039s)
    TimeoutError: Timeout reached on firefox on linux 4.15.0-1035-aws - mocha tests - run the mocha tests#
      at Timeout._onTimeout  <src/lib/Test.ts:221:24>
      at ontimeout  <timers.js:436:11>
      at tryOnTimeout  <timers.js:300:5>
      at listOnTimeout  <timers.js:263:5>
      at Timer.processTimers  <timers.js:223:10>
```

Going to restart...

**Comment by shane-tomlinson** [comment]
Different failures this time... :///////

**Pull request merged**